Can't notify() from config

```
$ git diff
diff --git a/alot/settings.py b/alot/settings.py
index a9cdb27..82d5ae7 100644
--- a/alot/settings.py
+++ b/alot/settings.py
@@ -188,6 +188,7 @@ class AlotConfigParser(FallbackConfigParser):
                                   " 'thread_highlight_rules' in section" \
                                   " 'general': {reason}".format(reason=err))
             logging.exception(report)
+            self.ui.notify('moo')
         finally:
             return rules

diff --git a/alot/ui.py b/alot/ui.py
index d3ada99..d5759cf 100644
--- a/alot/ui.py
+++ b/alot/ui.py
@@ -89,6 +89,7 @@ class UI(object):
         self.dbman.ui = self  # register ui with dbman
         self.logger = log
         self.accountman = accountman
+        config.ui = self # register ui with config parser

         if not colourmode:
             colourmode = config.getint('general', 'colourmode')

```

```
Traceback (most recent call last):
  File "/home/dtk/.local/bin/alot", line 20, in <module>
    main()
  File "/home/dtk/.local/lib/python2.7/site-packages/alot/init.py", line 172, in main
    args['colour-mode'],
  File "/home/dtk/.local/lib/python2.7/site-packages/alot/ui.py", line 113, in __init__
    self.mainloop.run()
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/main_loop.py", line 274, in run
    self.screen.run_wrapper(self._run)
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/raw_display.py", line 229, in run_wrapper
    return fn()
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/main_loop.py", line 285, in _run
    self.draw_screen()
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/main_loop.py", line 508, in draw_screen
    canvas = self._topmost_widget.render(self.screen_size, focus=True)
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/widget.py", line 132, in cached_render
    canv = fn(self, size, focus=focus)
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/widget.py", line 1318, in render
    canv = get_delegate(self).render(size, focus=focus)
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/widget.py", line 132, in cached_render
    canv = fn(self, size, focus=focus)
  File "/home/dtk/.local/lib/python2.7/site-packages/urwid/container.py", line 609, in render
    assert foot.rows() == frows, "rows, render mismatch"
AssertionError: rows, render mismatch
```

--- a/alot/settings.py
+++ b/alot/settings.py
@@ -87,6 +87,9 @@ class AlotConfigParser(FallbackConfigParser):
                 else:
                     self.set(section, key, value)
 
+        # parse highlighting rules
+        self._highlighting_rules = self._parse_highlighting_rules()
+
     def get_palette(self):
         """parse the sections '1c-theme', '16c-theme' and '256c-theme'
         into an urwid compatible colour palette.
@@ -174,7 +177,7 @@ class AlotConfigParser(FallbackConfigParser):
         has_bg = self.has_option(theme, themeing + '_bg')
         return (has_fg or has_bg)
 
-    def get_highlight_rules(self):
+    def _parse_highlighting_rules(self):
         """
         Parse the highlighting rules from the config file.
 
@@ -182,18 +185,23 @@ class AlotConfigParser(FallbackConfigParser):
         :rtype: :py:class:`collections.OrderedDict`
         """
         rules = OrderedDict()
+        config_string = self.get('highlighting', 'rules')
         try:
-            config_string = self.get('highlighting', 'rules')
             rules = json.loads(config_string, object_pairs_hook=OrderedDict)
-        except NoOptionError as err:
-            logging.exception(err)
         except ValueError as err:
-            report = ParsingError("Could not parse config option" \
-                                  " 'rules' in section 'highlighting':" \
-                                  " {reason}".format(reason=err))
-            logging.exception(report)
-        finally:
-            return rules
+            raise ParsingError("Could not parse config option" \
+                               " 'rules' in section 'highlighting':" \
+                               " {reason}".format(reason=err))
+        return rules
+
+    def get_highlight_rules(self):
+        """
+        Getter for the highlighting rules.
+
+        :returns: The highlighting rules
+        :rtype: :py:class:`collections.OrderedDict`
+        """
+        return self._highlighting_rules
 
     def get_tag_theme(self, tag, focus=False, highlight=''):
         """

